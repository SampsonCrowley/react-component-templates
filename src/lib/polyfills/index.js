import focusWithin from 'focus-within';
import './is-array';

focusWithin(document);
