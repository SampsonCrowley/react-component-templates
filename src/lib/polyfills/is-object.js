import './is-array';

if (!Object.isPureObject) {
  Object.isArray = Array.isArray;

  Object.isObject = function(arg, pure = false) {
    return !!(arg && (typeof arg === 'object') && !(pure && Object.isArray(arg)))
  };

  Object.isPureObject = function(arg, noClass = false) {
    return Object.isObject(arg, true) && !(noClass && Object.isClass(arg))
  };

  Object.isClass = function(arg) {
    return /object (?!Array|Object)[A-Za-z]+/.test(Object.prototype.toString.call(arg))
  }
}
