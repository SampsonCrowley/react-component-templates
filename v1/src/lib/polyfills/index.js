import focusWithin from 'focus-within';
import './filter';
import './includes';
import './is-array';
import './is-object';

focusWithin(document);
