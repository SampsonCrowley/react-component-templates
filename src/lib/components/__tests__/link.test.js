import 'polyfills'
import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Link from '../link'

describe('Components - Link', () => {
  const div = document.createElement('div');

  const createLink = ({...props}) => {
    ReactDOM.render((
      <Router>
        <Link {...props} />
      </Router>
    ), div);
    return div.querySelector('a')
  }

  it('renders an anchor tag', () => {
    const rendered = createLink({
      to: '/',
      children: 'Test'
    })
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("A")
    ReactDOM.unmountComponentAtNode(div);
  })

  it('requires a "to" prop', () => {
    const cons = global.console
    global.console = {error: jest.fn()}

    createLink({children: 'Test' })
    expect(console.error).toBeCalled()

    ReactDOM.unmountComponentAtNode(div);
    global.console = cons
  })

  it('sets the "href" prop to "to"', () => {
    expect(createLink({
      to: '/test',
      children: 'Test'
    }).href).toMatch(/http:\/{2}[A-Za-z09\.]+\/test$/)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('accepts mailto links', () => {
    expect(createLink({
      to: 'mailto:mail@mail.com',
      children: 'Test'
    }).href).toMatch(/^mailto/)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('accepts tel links', () => {
    expect(createLink({
      to: 'tel:+17777777777',
      children: 'Test'
    }).href).toMatch(/^tel/)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('accepts global links', () => {
    expect(createLink({
      to: '//test.com',
      children: 'Test'
    }).href).toMatch(/^(https?:)?\/\/test.com/)
    expect(createLink({
      to: 'https://test.com',
      children: 'Test'
    }).href).toMatch(/^https:\/\/test.com/)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('requires children', () => {
    const cons = global.console
    global.console = {error: jest.fn()}

    createLink({to: '/'})
    expect(console.error).toBeCalled()

    ReactDOM.unmountComponentAtNode(div);
    global.console = cons;
  })

  it('renders children', () => {
    expect(createLink({
      to: '/',
      children: 'Test'
    }).innerHTML).toMatch(/^Test$/)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('is snapshotable', () => {
    const tree = renderer
      .create(
        <Router>
          <Link to='/'>Test</Link>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
