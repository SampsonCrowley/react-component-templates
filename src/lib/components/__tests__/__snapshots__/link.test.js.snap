// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Link is snapshotable 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  Test
</a>
`;
